ecs-cli/modules/clients/aws/cloudformation/cluster_template.go

```diff
@@ -13,8 +13,65 @@
 
 package cloudformation
 
-func GetClusterTemplate() string {
-	return cluster_template
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+)
+
+func GetClusterTemplate(tags []*ecs.Tag, stackName string) (string, error) {
+	tagJSON, err := json.Marshal(tags)
+	if err != nil {
+		return "", err
+	}
+
+	asgTags := getASGTags(tags, stackName)
+	asgTagJSON, err := json.Marshal(asgTags)
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf(cluster_template, string(tagJSON), string(asgTagJSON)), nil
```

Comment:
String formatting on the template probably isn't extensible enough to be a long term solution. How verbose would it be to store the template as a struct(s) (custom or built into the SDK, as `ecs.Tag` is) which could be updated as such and then marshaled into JSON before returning? (probably a candidate for a separate PR, give the level of effort)

Not sure this is a blocker at this point but future edits/variability to the template will make this much more unwieldy. What do you think?

Reply:
This is a good idea; we should probably use: https://github.com/awslabs/goformation

It'd be a pretty decent sized refactoring though, so def would have to be a separate PR.